### PR TITLE
Multisocket support

### LINKED ISSUE #24: Plugin returns incorrect metrics, not taking into account multisocket platforms
In this section:
https://github.com/intelsdi-x/snap-plugin-collector-pcm/blob/8c4c9d8dda4d89d3fdcf4da77a94987dd0ce8536/pcm/pcm.go#L119

First line of output, which contains important information about contents of subsequent rows, is ignored.
The data is reported on system level but also per socket, giving something like this as output:
```
| System    | Socket 0  | Socket 1  |
|-----------|-----------|-----------|
|C1|C2|C3|C4|C1|C2|C3|C4|C1|C2|C3|C4|
|01|02|03|04|05|06|07|08|09|10|11|12|
```
In this case the plugin will report
`C1=9, C2=10, C3=11,C4=12`
which is not perceivable (I guess) on single socket platform, but unacceptable on multisocket. If metrics are not subdivided by sockets it should stop parsing after gathering system stats.

@intelsdi-x/plugin-maintainers

### PR DESCRIPTION
**Pull request by PatrykMatyjasek**
Fixes #24
- update tests
- update readme
- include parsing components names
- update examples

Signed-off-by: Patryk Matyjasek <patryk.matyjasek@intel.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,24 +155,7 @@ Create a task JSON file (exemplary file in examples/tasks/pcm-file.json):
...
-                "/intel/pcm/Proc_Energy_Joules": {},
-                "/intel/pcm/READ": {},
-                "/intel/pcm/SKT0": {},
-                "/intel/pcm/TEMP": {},
-                "/intel/pcm/TIME_ticks": {},
-                "/intel/pcm/WRITE": {}
+                "/intel/pcm/*": {}
             },
             "config": {
                 "/intel/pcm": {
```
Comment (IzabellaRaulin) [inline README.md]: Could you paste here the exemplary task manifest from examples/tasks? So, it would be consistent and what is more, this config section will be removed (there is no need to keep that for this plugin). Thank You

File: README.md [hunk 2]
```diff
@@ -191,28 +190,67 @@ State: Running
...
-/intel/pcm/L3MISS                0.415           2015-12-02 08:19:46.001144495 -0500 EST         gklab-108-166
-/intel/pcm/PhysIPC               4.33            2015-12-02 08:19:46.001145292 -0500 EST         gklab-108-166
-/intel/pcm/PhysIPC%              108             2015-12-02 08:19:46.001149828 -0500 EST         gklab-108-166
-/intel/pcm/Proc_Energy_Joules    8.46            2015-12-02 08:19:46.001145857 -0500 EST         gklab-108-166
-/intel/pcm/READ                  0.084           2015-12-02 08:19:46.00114662 -0500 EST          gklab-108-166
-/intel/pcm/SKT0                  8.46            2015-12-02 08:19:46.001152938 -0500 EST         gklab-108-166
-/intel/pcm/TEMP                  70              2015-12-02 08:19:46.001153401 -0500 EST         gklab-108-166
-/intel/pcm/TIME_ticks           3430            2015-12-02 08:19:46.001153955 -0500 EST         gklab-108-166
-/intel/pcm/WRITE                 0.0563          2015-12-02 08:19:46.00114718 -0500 EST          gklab-108-166
+$ snaptel task watch 44c01cd0-7133-49b1-a95c-a444db064b40
```
Comment (IzabellaRaulin) [inline README.md]: To be consistent, please update task ID in lines above - lines 182, 185, 186.

File: pcm/pcm_test.go [hunk 3]
```diff
@@ -100,7 +132,7 @@ func TestPCMPlugin(t *testing.T) {
 
 			So(func() { pcm.CollectMetrics(mockMts) }, ShouldNotPanic)
 			result, err := pcm.CollectMetrics(mockMts)
-			So(len(result), ShouldEqual, 24)
+			So(len(result), ShouldEqual, 56)
 			So(err, ShouldBeNil)
```
Comment (IzabellaRaulin) [inline pcm/pcm_test.go]: Could you change the order of So(...) - I mean checking if err is nil should be made as first.

File: pcm/pcm_test.go [hunk 4]
```diff
@@ -100,7 +132,7 @@ func TestPCMPlugin(t *testing.T) {
 
 			So(func() { pcm.CollectMetrics(mockMts) }, ShouldNotPanic)
 			result, err := pcm.CollectMetrics(mockMts)
-			So(len(result), ShouldEqual, 24)
+			So(len(result), ShouldEqual, 56)
```
Comment (IzabellaRaulin) [inline pcm/pcm_test.go]: I suggest to put here `So(len(result), ShouldEqual, len(refMap))` instead of  56 using explicitly.

**Comment by IzabellaRaulin** [comment]
Please increment plugin's version

File: README.md [hunk 5]
```diff
@@ -139,24 +155,7 @@ Create a task JSON file (exemplary file in examples/tasks/pcm-file.json):
...
-                "/intel/pcm/L3MISS": {},
-                "/intel/pcm/PhysIPC": {},
-                "/intel/pcm/PhysIPC%": {},
-                "/intel/pcm/Proc_Energy_Joules": {},
-                "/intel/pcm/READ": {},
-                "/intel/pcm/SKT0": {},
-                "/intel/pcm/TEMP": {},
-                "/intel/pcm/TIME_ticks": {},
-                "/intel/pcm/WRITE": {}
+                "/intel/pcm/*": {}
```
Comment (IzabellaRaulin) [inline README.md]: It seems that [examples/tasks/pcm-influxdb.json](https://github.com/intelsdi-x/snap-plugin-collector-pcm/blob/master/examples/tasks/pcm-influxdb.json) also should be updated.

File: README.md (hunk 2 above)
Comment (PatrykMatyjasek) [inline README.md]: done

File: pcm/pcm.go [hunk 6]
```diff
@@ -195,7 +204,8 @@ func (pcm *PCM) parse(reader io.Reader) {
 			for i, k := range keys[2:] {
 				// removes all spaces from metric key
 				metricKey := ns.ReplaceNotAllowedCharsInNamespacePart(k)
-				pcm.keys[i] = fmt.Sprintf("/intel/pcm/%s", metricKey)
+				metricComponent := ns.ReplaceNotAllowedCharsInNamespacePart(header[i+2])
```
Comment (IzabellaRaulin) [inline pcm/pcm.go]: What do you think about introducing a new var "fieldsToSkip=2", so it might be used in lines: 203, 204 and 207.  This might improve code readability.

File: pcm/pcm_test.go [hunk 7]
```diff
@@ -1,6 +1,5 @@
 //
 // +build small
-
```
Comment (IzabellaRaulin) [inline pcm/pcm_test.go]: Please revert this change:
```
pcm/pcm_test.go:2: +build comment must appear before package clause and be followed by a blank line
```
You can remove line 1 - it's not needed

File: README.md [hunk 8]
```diff
@@ -69,39 +69,55 @@ Customize path to pcm executable is also possible by setting environment variabl
...
+Metric	|	Explanation
+-----------	|	--------------
+EXEC	|	Instructions per nominal CPU cycle, i.e. in respect to the CPU frequency ignoring turbo and power saving
+IPC	|	Instructions per cycle. This measures how effectively you are using the core.
+FREQ	|	Frequency relative to nominal CPU frequency (“clockticks”/”invariant timer ticks”)
+AFREQ	|	Frequency relative to nominal CPU frequency excluding the time when the CPU is sleeping
+L3MISS	|	L3 cache line misses in millions
+L2MISS	|	L2 cache line misses in millions
+L3HIT	|	L3 Cache hit ratio (hits/reference)
+L2HIT	|	L2 Cache hit ratio (hits/reference)
```
Comment (IzabellaRaulin) [inline README.md]: Missing core residency metrics

File: README.md [hunk 9]
```diff
@@ -69,39 +69,55 @@ Customize path to pcm executable is also possible by setting environment variabl
...
+-----------	|	--------------
+READ	|	Memory read traffic on this socket in GB
+WRITE	|	Memory read traffic on this socket in GB
+
+The following metrics are only available on a socket level:
+
+Metric	|	Explanation
+-----------	|	--------------
+Proc Energy (Joules)	|	The energy consumed by the processor in Joules. Divide by the time to get the power consumption in watt
+DRAM Energy (Joules)	|	The energy consumed by the DRAM attached to this socket in Joules. Divide by the time to get the power consumption in watt
```
Comment (IzabellaRaulin) [inline README.md]: @PatrykMatyjasek - could you verify if DRAM Energy is exposed (I mean if executed pcm command is able to return that). Also, we should keep the same format as a metric name is exposed by plugin - there is metric `/intel/pcm/Proc_Energy_Joules/SKT0`, not `/intel/pcm/SKT0/Proc Energy (Joules)`.

File: README.md [hunk 10]
```diff
@@ -69,39 +69,40 @@ Customize path to pcm executable is also possible by setting environment variabl
...
+/intel/pcm/[Component]/FREQ	|	Frequency relative to nominal CPU frequency (“clockticks”/”invariant timer ticks”)
+/intel/pcm/[Component]/AFREQ	|	Frequency relative to nominal CPU frequency excluding the time when the CPU is sleeping
+/intel/pcm/[Component]/L3MISS	|	L3 cache line misses in millions
+/intel/pcm/[Component]/L2MISS	|	L2 cache line misses in millions
+/intel/pcm/[Component]/L3HIT	|	L3 Cache hit ratio (hits/reference)
+/intel/pcm/[Component]/L2HIT	|	L2 Cache hit ratio (hits/reference)
+/intel/pcm/[Component]/L3CLK	|	Very rough estimate of cycles lost to L3 cache misses vs. clockticks
+/intel/pcm/[Component]/L2CLK	|	Very rough estimate of cycles lost to L2 cache misses vs. clockticks
+/intel/pcm/[Component]/READ	|	Memory read traffic on this socket in GB
+/intel/pcm/[Component]/WRITE	|	Memory read traffic on this socket in GB
```
Comment (IzabellaRaulin) [inline README.md]: Bytes written to memory controller in GBytes.
